spring-cloud-stream/src/main/java/org/springframework/cloud/stream/config/BinderFactoryConfiguration.java

```diff
@@ -150,16 +156,19 @@ public BinderTypeRegistry binderTypeRegistry(ConfigurableApplicationContext conf
 		try {
 			Enumeration<URL> resources = classLoader.getResources("META-INF/spring.binders");
 			if (!Boolean.valueOf(this.selfContained) && (resources == null || !resources.hasMoreElements())) {
-				throw new BeanCreationException("Cannot create binder factory, no `META-INF/spring.binders` " +
-						"resources found on the classpath");
+				this.logger.warn("Failed to locate 'META-INF/spring.binders' resources on the classpath."
+						+ " Assuming standard boot 'META-INF/spring.factories' configuration is used");
```

Comment:
When we fall back to the `spring.binders`check, that means that we haven't found any binders yet. In that check, if we still don't find any binder, don't we want to throw an exception?  This change only logs the warning and let the application comes up without a binder implementation present. Is that intended?

Reply:
Yes, the issue is that the above code executes way to early to create `BinderTypeRegistry` bean (so AC initialization), hence the warn message.

So basically we are letting it through with the WARN and then in `DefaultBinderFactory.getBinder(..)` is where we try to find binder in the current AC and if not found we fall back on `doGetBinder(..)` which is the old method (unchanged) where we go and create child AC.

But that makes me think if we should add that same check right before `DefaultBinderFactory.doGetBinder(..)`? It would be easy, but there are already two assertions in that method so it will fail properly. 
WDYT?

Reply:
Do we even need this warning at all? It's noise. Maybe DEBUG at most is all that's needed since we'll detect the error later. "no beans and no spring.binders".

Reply:
@olegz I wasn't thinking about the earlier execution phase you mentioned. I am fine either way then. As @garyrussell mentioned, we might want to change the logging to `DEBUG` to reduce potential noise.

Reply:
@sobychacko if that's the only change then please change it to DEBUG during merge.